python/cudf/cudf/core/column/datetime.py

```diff
@@ -630,6 +630,38 @@ def quantile(
             )
         return result.astype(self.dtype)
 
+    def find_and_replace(
+        self,
+        to_replace: ColumnBase,
+        replacement: ColumnBase,
+        all_nan: bool = False,
+    ) -> DatetimeColumn:
+        if not isinstance(to_replace, DatetimeColumn):
+            to_replace = cudf.core.column.as_column(to_replace)
+            if to_replace.can_cast_safely(self.dtype):
+                to_replace = to_replace.astype(self.dtype)
+        if not isinstance(replacement, DatetimeColumn):
+            replacement = cudf.core.column.as_column(replacement)
+            if replacement.can_cast_safely(self.dtype):
+                replacement = replacement.astype(self.dtype)
+        if isinstance(to_replace, DatetimeColumn):
+            to_replace = to_replace.as_numerical_column(
+                dtype=np.dtype("int64")
+            )
+        if isinstance(replacement, DatetimeColumn):
+            replacement = replacement.as_numerical_column(
+                dtype=np.dtype("int64")
+            )
+        try:
+            result_col = (
+                self.as_numerical_column(dtype=np.dtype("int64"))
+                .find_and_replace(to_replace, replacement, all_nan)
+                .astype(self.dtype)
+            )
+        except TypeError:
```

Comment:
What case is this and why are we OK with simply returning self?

Reply:
If there are any type casting issue for a mismatching column passed like `datatime.find_and_replace(to_replace=StringColumn, replacement=NumericalColumn)` we should just return the copy of the original column. This is a common pattern if we do `df.replace(...)` as most of the times all the inputs of `to_replace` & `replacement` are passed to columns of mismatching types.